src/main/java/frc/robot/swerve/SwerveDrive.java

```diff
@@ -27,18 +27,29 @@ public class SwerveDrive extends SubsystemBase {
     private final SwerveDriveInputsAutoLogged loggerInputs = new SwerveDriveInputsAutoLogged();
     private final SwerveModuleState[] currentModuleStates = new SwerveModuleState[4];
 
-    private SwerveDrive(boolean isReal,
+    private SwerveDrive(boolean isReal, boolean isNeo,
+                        boolean[] drivesInverted, boolean[] anglesInverted,
                         int[] driveIds,
                         int[] angleIds,
                         int[] encoderIds) {
         if (isReal) {
             for (int i = 0; i < modules.length; i++) {
-                ModuleIO io = new ModuleIOReal(
-                        driveIds[i],
-                        angleIds[i],
-                        encoderIds[i],
-                        SwerveConstants.motionMagicConfigs[i],
-                        i + 1);
+                ModuleIO io;
+                if (!isNeo) {
```

Comment:
Better to do `if (isNeo)`, direct comparison is better than inverted one.

Reply:
I was thinking once we merge the phoenix and this pr we pass the ModuleIOs and the GyroIO instead.

Reply:
Oh great